addons/mail/static/src/discuss/core/web/discuss_sidebar_categories_patch.xml

```diff
@@ -1,11 +1,35 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <templates xml:space="preserve">
+    <t t-inherit="mail.DiscussSidebarCategory" t-inherit-mode="extension">
+        <xpath expr="//*[@name='dropdownCategoryName']" position="replace">
+            <AutoresizeInput
+                t-if="state.renaming"
+                className="'fw-bold text-uppercase small user-select-none'"
+                enabled="true"
+                autofocus="true"
+                value="category.name"
+                onValidate.bind="renameCategory"
+            />
+            <span t-else="" class="fw-bold text-uppercase small user-select-none" t-esc="category.name"/>
+        </xpath>
+    </t>
     <t t-inherit="mail.DiscussSidebarCategory.main" t-inherit-mode="extension">
         <xpath expr="//*[@name='header']" position="after">
             <div t-if="state.editing and !store.discuss.isSidebarCompact" class="p-2" t-ref="selector">
                 <ChannelSelector category="category" onValidate.bind="stopEditing" autofocus="true" close.bind="stopEditing" class="'p-1'"/>
             </div>
         </xpath>
+        <xpath expr="//*[@name='categoryName']" position="replace">
+            <AutoresizeInput
+                t-if="state.renaming"
+                className="`btn-sm text-uppercase text-break fw-bolder smaller ${ !category.open and 'opacity-50' } ${ state.isRenaming and 'opacity-100' }`"
+                enabled="true"
+                autofocus="true"
+                value="category.name"
+                onValidate.bind="renameCategory"
+            />
+            <span t-else="" class="btn-sm p-0 text-uppercase text-break text-truncate fw-bolder smaller" t-att-class="{ 'opacity-50': !category.open }" t-esc="category.name" />
+        </xpath>
```

Comment:
I would remove the renaming of category from sidebar, this bloated the UI too much for the main screen of Discuss.

Editing categories and mapping a channel to a given category should all be done in advanced screens, usually through odoo views like the Channel settings form view page.